src/main/java/com/ita/if103java/ims/dao/impl/DashboardDaoImpl.java

```diff
@@ -175,16 +175,21 @@ private CRUDException crudException(String message, String operation, String att
                 "LIMIT ? ";
 
         static final String SQL_FIND_WAREHOUSE_LOAD_BY_ACCOUNT_ID =
-            "SELECT wcap.id, ifnull(charge,0) charge, capacity " +
-                "FROM " +
-                "(SELECT wh.top_warehouse_id id, sum(capacity) capacity, account_id, active " +
-                "FROM warehouses wh WHERE account_id=? and active=1 GROUP BY wh.top_warehouse_id) AS wcap " +
-                "LEFT OUTER JOIN " +
-                "(SELECT top_warehouse_id id, sum(quantity*volume) charge " +
-                "FROM saved_items si, items it, warehouses wh " +
-                "WHERE si.item_id=it.id and si.warehouse_id=wh.id " +
-                "GROUP BY top_warehouse_id) AS wcharge " +
-                "ON wcap.id=wcharge.id ";
+            "SELECT top_warehouse_id id, ifnull(sum(capacity),0) capacity, ifnull(sum(charge),0) charge FROM " +
+                "(SELECT warehouse_id, it.account_id, sum(quantity*volume) charge " +
+                "FROM saved_items si " +
+                "JOIN items it " +
+                "ON si.item_id = it.id " +
+                "JOIN warehouses wh " +
+                "ON si.warehouse_id = wh.id " +
+                "GROUP BY warehouse_id) AS  cha " +
+                "RIGHT JOIN " +
+                "(SELECT id, capacity, account_id, top_warehouse_id " +
+                "FROM warehouses " +
+                "WHERE is_bottom=1) AS cap " +
+                "ON  cha.warehouse_id=cap.id AND  cha.account_id=cap.account_id " +
+                "WHERE cap.account_id=? " +
+                "GROUP BY top_warehouse_id";
```

Comment:
again change) 
I would like to switch subqueries and use LEFT JOIN, as it was before. If it possible, sure.
Please just reply to me and not change

Reply:
After I delete recursive select, I thought it'll be better to make this one look like other and even if this version is bigger, isn't it more readable